src/redux/slice/userSlice.ts

```diff
@@ -0,0 +1,195 @@
+  'user/authGoogle',
+  async (login, thunkAPI) => {
+    try {
+      const user = await signInWithGoogle(login);
+      return user;
+    } catch (error: unknown) {
+      return thunkAPI.rejectWithValue(error);
+    }
+  },
+);
+
+export const updateProfile = createAsyncThunk(
+  'auth/updateProfile',
+  async (
+    { user, firstName, lastName }: { user: User; firstName: string; lastName: string },
+    thunkAPI,
+  ) => {
+    try {
+      const updatedUser = await updateUserProfile(user, firstName, lastName);
+      return updatedUser;
+    } catch (error: unknown) {
+      return thunkAPI.rejectWithValue(error);
+    }
+  },
+);
+export const updateLogin = createAsyncThunk(
+  'auth/updateLogin',
+  async ({ user, newLogin }: { user: User; newLogin: string }, thunkAPI) => {
+    try {
+      const updatedUser = await updateUserLogin(user, newLogin);
+      return updatedUser;
+    } catch (error: unknown) {
+      return thunkAPI.rejectWithValue(error);
+    }
+  },
+);
+
+export const uploadAvatar = createAsyncThunk(
+  'auth/uploadAvatar',
+  async (file: File, thunkAPI) => {
+    const state = thunkAPI.getState() as RootState;
+    try {
+      const url = await uploadFile(state.user, file, state.user.id);
+      return url;
+    } catch (error: unknown) {
+      return thunkAPI.rejectWithValue(error);
+    }
+  },
+);
+
+export const deleteUserAvatar = createAsyncThunk(
+  'user/deleteAvatar',
+  async (avatarUrl: string) => {
+    await deleteAvatar(avatarUrl);
+    return avatarUrl;
+  },
+);
+
+export const logout = createAsyncThunk('user/logout', async () => {
+  await firebaseLogout();
+  return {};
+});
+
+const authSlice = createSlice({
+  name: 'auth',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(register.fulfilled, (state, action) => {
+        if (action.payload) {
+          state.status = 'succeeded';
+          state.user = action.payload;
+          state.isAuthenticated = true;
+        }
+      })
+      .addCase(login.fulfilled, (state, action) => {
+        if (action.payload) {
+          state.status = 'succeeded';
+          state.user = action.payload;
+          state.role = action.payload.role;
+          state.isAuthenticated = true;
+        }
+      })
+      .addCase(authGoogle.fulfilled, (state, action) => {
+        if (action.payload) {
+          state.status = 'succeeded';
+          state.user = action.payload;
+          state.isAuthenticated = true;
+        }
+      });
+    builder.addCase(updateLogin.fulfilled, (state, action) => {
+      if (action.payload) {
+        state.status = 'succeeded';
+        state.user = action.payload;
+        state.isAuthenticated = true;
+      }
+    });
+    builder.addCase(updateProfile.fulfilled, (state, action) => {
+      if (action.payload) {
+        state.status = 'succeeded';
+        state.user = action.payload;
+        state.isAuthenticated = true;
+      }
```

Comment:
try use addMacher for recurring actions and use:
```suggestion
      .addCase(authGoogle.fulfilled, (state, action) => {
        if (action.payload) {
          state.status = 'succeeded';
          state.user = action.payload;
          state.isAuthenticated = true;
        }
      })
    .addCase(updateLogin.fulfilled, (state, action) => {
      if (action.payload) {
        state.status = 'succeeded';
        state.user = action.payload;
        state.isAuthenticated = true;
      }
    })
    .addCase(updateProfile.fulfilled, (state, action) => {
      if (action.payload) {
        state.status = 'succeeded';
        state.user = action.payload;
        state.isAuthenticated = true;
      }
```